src/ai/backend/common/validators.py

```diff
@@ -657,19 +655,19 @@ def __init__(
         self.scheme_required = scheme_required
 
     def check_and_return(self, value: Any) -> yarl.URL:
-        if not isinstance(value, (str, bytes)):
-            self._failure("A URL must be a unicode string or a byte sequence", value=value)
         if isinstance(value, bytes):
             value = value.decode("utf-8")
-        if self.scheme_required:
-            if not self.rx_scheme.match(value):
-                self._failure(
-                    "The given value does not have the scheme (protocol) part", value=value
-                )
         try:
-            return yarl.URL(value)
-        except ValueError as e:
-            self._failure(f"cannot convert the given value to URL (error: {e!r})", value=value)
+            parsed_url = yarl.URL(value)
+            if self.scheme_required:
+                if parsed_url.scheme == "":
+                    self._failure(
+                        "The given value does not have the scheme (protocol) part", value=value
+                    )
+        except Exception as e:
+            self._failure(f"{e}", value=value)
```

Comment:
```suggestion
            self._failure(repr(e), value=value)
```
this would be better